application/src/main/java/org/opentripplanner/graph_builder/module/osm/EscalatorProcessor.java

```diff
@@ -64,36 +54,21 @@ public void buildEscalatorEdge(OsmWay escalatorWay, double length) {
         );
       }
     }
-    for (int i = 0; i < nodes.size() - 1; i++) {
-      if (escalatorWay.isForwardEscalator()) {
-        EscalatorEdge.createEscalatorEdge(
-          intersectionNodes.get(nodes.get(i)),
-          intersectionNodes.get(nodes.get(i + 1)),
-          length,
-          duration.orElse(null)
-        );
-      } else if (escalatorWay.isBackwardEscalator()) {
-        EscalatorEdge.createEscalatorEdge(
-          intersectionNodes.get(nodes.get(i + 1)),
-          intersectionNodes.get(nodes.get(i)),
-          length,
-          duration.orElse(null)
-        );
-      } else {
-        EscalatorEdge.createEscalatorEdge(
-          intersectionNodes.get(nodes.get(i)),
-          intersectionNodes.get(nodes.get(i + 1)),
-          length,
-          duration.orElse(null)
-        );
-
-        EscalatorEdge.createEscalatorEdge(
-          intersectionNodes.get(nodes.get(i + 1)),
-          intersectionNodes.get(nodes.get(i)),
-          length,
-          duration.orElse(null)
-        );
-      }
+    if (escalatorWay.isForwardEscalator()) {
+      return new EscalatorEdgePair(
+        EscalatorEdge.createEscalatorEdge(fromVertex, toVertex, length, duration.orElse(null)),
+        null
+      );
+    } else if (escalatorWay.isBackwardEscalator()) {
```

Comment:
You can just use if here since the previous if returns now and you don't need to wrap the last return with else.

Reply:
I changed it to the suggestion